src/main/java/uk/gov/digital/ho/hocs/casework/reports/domain/mapping/UserNameValueMapper.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.digital.ho.hocs.casework.reports.domain.mapping;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.UserDto;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.EVENT;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.REPORT_MAPPER_USER_CACHE_ERROR;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.REPORT_MAPPER_USER_CACHE_REFRESH;
+
+@Slf4j
+@Service
+@Profile("reports")
+public class UserNameValueMapper implements ReportValueMapper<UUID, String> {
+
+    private final InfoClient infoServiceClient;
+
+    private final LoadingCache<UUID, String> uuidToUserNameCache;
+
+    @Autowired
+    public UserNameValueMapper(InfoClient infoServiceClient) {
+        this.infoServiceClient = infoServiceClient;
+        this.uuidToUserNameCache = Caffeine.newBuilder().build(this::fetchUserNameByUUID);
+        refreshCache();
+    }
+
+    private String fetchUserNameByUUID(UUID userUUID) {
+        try {
+            UserDto user = infoServiceClient.getUser(userUUID);
+            return formatUsername(user);
+        } catch (Exception e) {
+            log.warn("Failed to fetch user with UUID {}", userUUID, REPORT_MAPPER_USER_CACHE_ERROR);
+            return null;
```

Comment:
I wonder if we would be better returning the UUID here? Returning null might imply the case isn't assigned to a user so while showing the user the UUID isn't ideal it at least maintains the expected functionality.